changeset:   623945:6117c9ecd16b
tag:         tip
parent:      623941:45e313943df5
user:        stransky <stransky@redhat.com>
date:        Fri Jun 17 12:36:38 2022 +0200
files:       gfx/thebes/gfxPlatform.cpp
description:
Bug 1774271 [Linux] Don't use EGL_MESA_image_dma_buf_export in Mesa/Intel due to https://gitlab.freedesktop.org/mesa/mesa/-/issues/6688 r?jgilbert

Depends on https://phabricator.services.mozilla.com/D149238

Differential Revision: https://phabricator.services.mozilla.com/D149608


diff --git a/gfx/thebes/gfxPlatform.cpp b/gfx/thebes/gfxPlatform.cpp
--- a/gfx/thebes/gfxPlatform.cpp
+++ b/gfx/thebes/gfxPlatform.cpp
@@ -2871,6 +2871,12 @@ void gfxPlatform::InitWebGLConfig() {
         adapterDriverVendor.Find("radeonsi") != -1) {
       gfxVars::SetUseDMABufSurfaceExport(false);
     }
+    // Disable EGL_MESA_image_dma_buf_export on mesa/iris due to
+    // https://gitlab.freedesktop.org/mesa/mesa/-/issues/6688
+    if (adapterDriverVendor.Find("mesa") != -1 &&
+        adapterDriverVendor.Find("iris") != -1) {
+      gfxVars::SetUseDMABufSurfaceExport(false);
+    }
   }
 }
 

